tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/symbols/SymbolsInResourceMethodsTest.java

```diff
@@ -0,0 +1,176 @@
+import static io.ballerina.semantic.api.test.util.SemanticAPITestUtils.getDefaultModulesSemanticModel;
+import static io.ballerina.semantic.api.test.util.SemanticAPITestUtils.getDocumentForSingleSource;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Test cases for symbols in resource methods.
+ *
+ * @since 2201.6.0
+ */
+public class SymbolsInResourceMethodsTest {
+
+    private SemanticModel model;
+    private Document srcFile;
+
+    @BeforeClass
+    public void setup() {
+        Project project = BCompileUtil.loadProject("test-src/symbols/resource_method_symbols.bal");
+        model = getDefaultModulesSemanticModel(project);
+        srcFile = getDocumentForSingleSource(project);
+    }
+
+    @Test(dataProvider = "SymbolInResourceMethodPosProvider")
+    public void testSymbolsInResourceMethod(int line, int col, String name, SymbolKind kind) {
+        assertBasicsAndGetSymbol(model, srcFile, line, col, name, kind);
+    }
+
+    @DataProvider(name = "SymbolInResourceMethodPosProvider")
+    public Object[][] getSymbolPos() {
+        return new Object[][]{
+                {17, 22, "get", SymbolKind.RESOURCE_METHOD},
+                {24, 23, "get", SymbolKind.RESOURCE_METHOD},
+                {24, 26, "authors", SymbolKind.PATH_SEGMENT},
+                {28, 42, "author", SymbolKind.PARAMETER},
+                {24, 45, "names", SymbolKind.PATH_PARAMETER},
+                {32, 52, "isbn", SymbolKind.PATH_PARAMETER},
+                {32, 80, "Book", SymbolKind.TYPE},
+
+                // TODO: The following tests need to be verified again
+                {17, 26, ".", SymbolKind.PATH_PARAMETER},   // TODO: Is this a path-param?
+                {36, 27, "$^", SymbolKind.PATH_PARAMETER},
+                {40, 36, "$^^", SymbolKind.PATH_PARAMETER},
+
+        };
+    }
+
+    @Test(dataProvider = "PathParamPosProvider")
+    public void testPathParamSymbols(int line, int col, String name, PathSegment.Kind pathKind, TypeDescKind typekind) {
+        PathParameterSymbol symbol = (PathParameterSymbol) assertBasicsAndGetSymbol(model, srcFile, line, col,
+                                                                                    name, SymbolKind.PATH_PARAMETER);
+        assertEquals(symbol.pathSegmentKind(), pathKind);
+
+        TypeSymbol typeDescriptor = symbol.typeDescriptor();
+        assertEquals(typeDescriptor.typeKind(), typekind);
+    }
+
+    @DataProvider(name = "PathParamPosProvider")
+    public Object[][] getPathParamPos() {
+        return new Object[][]{
+                {24, 45, "names", PathSegment.Kind.PATH_REST_PARAMETER, TypeDescKind.ARRAY},
+                {32, 52, "isbn", PathSegment.Kind.PATH_PARAMETER, TypeDescKind.UNION},
+                {36, 48, "author", PathSegment.Kind.PATH_PARAMETER, TypeDescKind.STRING},
+                {36, 61, "bookId", PathSegment.Kind.PATH_PARAMETER, TypeDescKind.INT},
```

Comment:
If the `pathKind` is always the same, can avoid it being passed from the data provider.

```suggestion
    public void testPathParamSymbols(int line, int col, String name, TypeDescKind typekind) {
        PathParameterSymbol symbol = (PathParameterSymbol) assertBasicsAndGetSymbol(model, srcFile, line, col,
                                                                                    name, SymbolKind.PATH_PARAMETER);
        assertEquals(symbol.pathSegmentKind(), PathSegment.Kind.PATH_PARAMETER);

        TypeSymbol typeDescriptor = symbol.typeDescriptor();
        assertEquals(typeDescriptor.typeKind(), typekind);
    }

    @DataProvider(name = "PathParamPosProvider")
    public Object[][] getPathParamPos() {
        return new Object[][]{
                {24, 45, "names", TypeDescKind.ARRAY},
                {32, 52, "isbn", TypeDescKind.UNION},
                {36, 48, "author", TypeDescKind.STRING},
                {36, 61, "bookId", TypeDescKind.INT},
```

Reply:
Here the path kind is not the same. We have two path kinds:
1. PATH_REST_PARAMETER
2. PATH_PARAMETER